src/components/Main/Main.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useContext, useState } from 'react';
+
+import './Main.scss';
+import { TodoList } from '../TodoList';
+import { TodosContext } from '../TodosContext';
+import { Status } from '../../types/Status';
+import { Todo } from '../../types/Todo';
+
+export const Main: React.FC = () => {
+  const { todos, dispatch, currentFilter } = useContext(TodosContext);
+
+  const [isChecked, setIsChecked] = useState(false);
+
+  let filteredTodos: Todo[] = [];
+
+  switch (currentFilter) {
+    case Status.All:
+      filteredTodos = [...todos];
+      break;
+
+    case Status.Active:
+      filteredTodos = todos.filter(todo => !todo.completed);
+      break;
+
+    case Status.Completed:
+      filteredTodos = todos.filter(todo => todo.completed);
+      break;
+
+    default:
+      throw new Error('Unknown filter!');
+  }
```

Comment:
My assumption is move this logic into helper function to get visible todos it will make your component cleaner
`const visibleTodos = getVisibleTodos(todos, currentFilter);`